### PR TITLE
Added new function for using HTTPS when checking to see if a bucket e…

### PR DESCRIPTION
**Pull request by shawnallen85**
Added a new function to see if a bucket exists, uses HTTPS rather than HTTP.

## Motivation and Context
Security group won't allow HTTP protocol to validate S3 bucket.

## Testing
Sadly I didn't see any unit tests for the original function (DoesS3BucketExistAsync) so I did not add any that are specific to this.

## Screenshots (if appropriate)
N/A

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [X] My code follows the code style of this project
- [X] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [X] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Comment by shawnallen85** [comment]
Resolves issue https://github.com/aws/aws-sdk-net/issues/1107. I hope.

File: sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs [hunk 1]
```diff
@@ -616,5 +615,81 @@ public static async System.Threading.Tasks.Task<bool> DoesS3BucketExistAsync(IAm
...
+        /// well as Signature V4-optional regions, we keep the expiry to within the maximum for V4 
+        /// (which is one week).
+        /// 2. Making a HEAD request to the Url
+        /// </summary>
+        /// <param name="bucketName">The name of the bucket to check.</param>
+        /// <param name="s3Client">The Amazon S3 Client to use for S3 specific operations.</param>
+        /// <returns></returns>
+        public static async System.Threading.Tasks.Task<bool> DoesS3BucketExistV2Async(IAmazonS3 s3Client, string bucketName)
+        {
+            if (s3Client == null)
```
Comment (dermeister0) [inline sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs]: A lot of copy-paste code.

**Comment by dermeister0** [comment]
Why do we need a new function? I think need to fix the existing one.

File: sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs (hunk 1 above)
Comment (shawnallen85) [reply sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs]: Removed duplication of code

**Comment by shawnallen85** [comment]
@dermeister0 Reason V2 was added is due to this comment: https://github.com/aws/aws-sdk-net/issues/1107#issuecomment-437535437

**Comment by shawnallen85** [comment]
@dermeister0 Let me know if that change is acceptable or not. Thanks!

File: sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs [hunk 2]
```diff
@@ -616,5 +634,23 @@ public static async System.Threading.Tasks.Task<bool> DoesS3BucketExistAsync(IAm
...
+        /// This is done by:
+        /// 1. Creating a PreSigned Url for the bucket. To work with Signature V4 only regions, as
+        /// well as Signature V4-optional regions, we keep the expiry to within the maximum for V4 
+        /// (which is one week).
+        /// 2. Making a HEAD request to the Url
+        /// </summary>
+        /// <param name="bucketName">The name of the bucket to check.</param>
+        /// <param name="s3Client">The Amazon S3 Client to use for S3 specific operations.</param>
+        /// <returns></returns>
+        public static async System.Threading.Tasks.Task<bool> DoesS3BucketExistV2Async(IAmazonS3 s3Client, string bucketName)
```
Comment (matteo-prosperi) [inline sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs]: `async` is not superfluous here, you can just remove `await` below.

File: sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs [hunk 3]
```diff
@@ -554,6 +553,25 @@ internal static bool IsInstructionFile(string key)
         /// <param name="s3Client">The Amazon S3 Client to use for S3 specific operations.</param>
         /// <returns></returns>
         public static async System.Threading.Tasks.Task<bool> DoesS3BucketExistAsync(IAmazonS3 s3Client, string bucketName)
```
Comment (matteo-prosperi) [inline sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs]: `async` is not superfluous here, you can just remove `await` below.

**Comment by dpope-ea** [comment]
Hello all, any update on when this will be merged to aws:master and a new SDK package released to nuget?

**Comment by matteo-prosperi** [comment]
Hello,
AWSSDK.S3 version 3.3.104.0 contains a DoesBucketExistsV2 version which addresses this issue (https://github.com/aws/aws-sdk-net/commit/208377db24731da445d23da7dd8bea60e33a4f17).

**Pull request closed without merging**